algorithms/CPlusPlus/Stacks/stack-using-array.cpp

```diff
@@ -0,0 +1,137 @@
+/* Program to implement stack using array */
+#include<iostream>
+using namespace std;
+class Stack
+{
+    int cur ,capacity ,*arr;
+public:
+    // constructor which takes capacity as argument
+    Stack(int capacity)
+    {
+        this->capacity = capacity;
+        cur = -1;
+        // dynamically allocating memory
+        arr=new int[capacity];
+        // if arr contain null
+        if(arr == NULL)
+        {
+            cout << "Memory not located";
+            exit(0);
+        }
+    }
+    ~Stack()
+    {
+        delete arr;
```

Comment:
Using wrong delete expression would lead to undefined behaviour in your program.

```suggestion
        delete[] arr;
```